src/Generator.Rewrite/Method/MethodConstructor_Debug.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
+
+namespace OpenTK.Rewrite.Method
+{
+    public partial class MethodConstructor
```

Comment:
I don't think a partial class is neccessarily the right choice here. A deeper refactoring of the MethodGenerator class to reduce its size might be better in the long term.

Reply:
I've been thinking about a good approach for this... I don't want it to be as tightly coupled as it currently is.

I've been thinking about some level of IoC, but am not sure if it'd be a bit overengineered for this purpose...
An example/starting point to this idea can be found [here](https://github.com/FreezyLemon/opentk/compare/4.0-refactor-rewrite...FreezyLemon:4.0-rewrite-new-approach) (take a look at `MethodRewriter`, the rest of the code is nearly the same).

I'd love feedback on my idea or just suggestions in general!